### ISSUE TITLE
[`delete_file.lua` Bug report] `MoveToFolder` not working unless `DeletedFilesPath` is specified and `DeletedFilesPath` is in the same dir as the videos

### ISSUE THREAD

**Issue by kohane27** [zenyd/mpv-scripts#44]
TL;DR: 

1. Videos aren't moved unless `DeletedFilesPath` is specified and `DeletedFilesPath` is in the same dir as the videos
2. I can't move videos from different directories, i.e., I can move videos from `~/Videos` but not `/run/media/username/MX500/`

## Description

Hello there. Hope you're doing well.

Thank you for creating this handy tool. I finally don't have to use a file manager to find the video and then delete it. However, I've encountered a bug:

## Scenario 1 (fail)

`delete_file.conf`:

```
MoveToFolder=yes
```

Then:

```
❯ mpv "test-delete.webm"
   cplayer:  (+) Video --vid=1 (*) (vp9 1920x1080 23.976fps)
   cplayer:  (+) Audio --aid=1 --alang=eng (*) (opus 2ch 48000Hz)
Cannot load libcuda.so.1
   cplayer: Saving state.
   cplayer:
   cplayer: Exiting... (Quit)
delete_file: moving: /home/username/Videos/test-delete.webm
```

I can see the dir `~/delete_file` is created, but `test-delete.webm` isn't moved there.

## Scenario 2 (success)

`delete_file.conf`:

```
MoveToFolder=yes
DeletedFilesPath=/home/username/Videos/delete_file
```

```
❯ mpv "test-delete.webm"
   cplayer:  (+) Video --vid=1 (*) (vp9 1920x1080 23.976fps)
   cplayer:  (+) Audio --aid=1 --alang=eng (*) (opus 2ch 48000Hz)
Cannot load libcuda.so.1
   cplayer: Saving state.
   cplayer:
   cplayer: Exiting... (Quit)
delete_file: moving: /home/username/Videos/test-delete.webm
```

The dir `/home/username/Videos/delete_file` is created and `test-delete.webm` is moved there.

## Scenario 3 (fail)

`delete_file.conf`:

```
MoveToFolder=yes
DeletedFilesPath=/home/username/Videos/delete_file
```

Note: this `test-delete.webm` is in `/run/media/username/MX500`, so different than `~/Videos` above.

```
❯ mpv "test-delete.webm"
   cplayer:  (+) Video --vid=1 (*) (h264 1920x1080 30.000fps)
   cplayer:  (+) Audio --aid=1 (*) (aac 2ch 24000Hz)
Cannot load libcuda.so.1
progressbar: Playing: "test-delete.webm"
   cplayer: Saving state.
   cplayer:
   cplayer: Exiting... (Quit)
delete_file: moving: /run/media/username/MX500/test-delete.webm
```

The video `test-delete.webm` is not moved anywhere.

## Scenario 4 (fail)

`delete_file.conf`:

```
MoveToFolder=yes
```

Note: this `test-delete.webm` is again, in `/run/media/username/MX500`.

```
❯ mpv "test-delete.webm"
   cplayer:  (+) Video --vid=1 (*) (h264 1920x1080 30.000fps)
   cplayer:  (+) Audio --aid=1 (*) (aac 2ch 24000Hz)
Cannot load libcuda.so.1
   cplayer: Saving state.
   cplayer:
   cplayer: Exiting... (Quit)
delete_file: moving: /run/media/username/MX500/test-delete.webm
```

The video `test-delete.webm` is not moved anywhere as well.

## Scenario 5 (success)

`delete_file.conf`:

```
MoveToFolder=yes
DeletedFilesPath=/run/media/username/MX500/delete_file/
```

Note: this `test-delete.webm` is in `/run/media/username/MX500`, so different than `~/Videos` above.

```
❯ mpv "test-delete.webm"
   cplayer:  (+) Video --vid=1 (*) (h264 1920x1080 30.000fps)
   cplayer:  (+) Audio --aid=1 (*) (aac 2ch 24000Hz)
Cannot load libcuda.so.1
   cplayer: Saving state.
   cplayer:
   cplayer: Exiting... (Quit)
delete_file: moving: /run/media/username/MX500/test-delete.webm
```

The video `test-delete.webm` is indeed moved to `/run/media/username/MX500/delete_file`

## Conclusion

However, if I specify `MoveToFolder=no`, then videos are deleted across all mount points. But this is not what I want: I want to move videos to a directory across all mount points. I suspect this is a permission to move files issue? Because given `Scenario 1`, the video can't be moved to `~/delete_file` even it's the same mount point (`/home/username`).

Config:

```
❯ mpv --version
mpv 0.34.1-dirty Copyright © 2000-2021 mpv/MPlayer/mplayer2 projects
FFmpeg version: n5.0.1

❯ uname
Linux
```

Any input is much appreciated. Thank you:)

**Comment by zenyd** [comment]
I haven't tested these scenarios, but noticed that the player is saving state so it can resume where left of? Is this right? If this is enabled by default maybe try disabling this.

**Comment by zenyd** [comment]
I tested this myself on my linux machine and could not reproduce. MoveToFolder moves the files as expected. I assume it is a permission problem. If you have any interest in debugging this you can reopen the issue
